perf.go

```diff
@@ -120,7 +124,15 @@ func (m *PerfMap) Start() error {
 				}
 				continue
 			}
-			m.DataHandler(record.CPU, record.RawSample, m, m.manager)
+			if m.RecordHandler != nil {
+				// it is expected that we maintain ownership of `record` and can reuse once the callback returns
+				m.RecordHandler(&record, m, m.manager)
+			} else if m.DataHandler != nil {
+				// maintain previous semantics of getting a copy of the data
+				buf := make([]byte, len(record.RawSample))
+				copy(buf, record.RawSample)
```

Comment:
do we really need to copy here ?

Reply:
This was done to preserve the expectation that previous clients would have about the slice. Previously it was a copy and was safe to hold on to.

Reply:
Thanks for the explanation